tests/integration/test_integration.py

```diff
@@ -986,3 +995,34 @@ def test_no_warnings():
     out, err, _, _ = call_program("ddtrace-run", sys.executable, "-Wall", "-c", "'import ddtrace'", env=env)
     assert out == b"", out
     assert err == b"", err
+
+
+def test_civisibility_event_endpoints():
+    with override_env(dict(DD_API_KEY="foobar.baz")):
+        with override_global_config({"_ci_visibility_code_coverage_enabled": True}):
+            t = Tracer()
+            t.configure(writer=CIVisibilityWriter(reuse_connections=True))
+            t._writer._conn = mock.MagicMock()
+            with mock.patch("ddtrace.internal.writer.Response.from_http_response") as from_http_response:
+                from_http_response.return_value.status = 200
+                s = t.trace("operation", service="svc-no-cov")
+                s.finish()
+                span = t.trace("operation2", service="my-svc2")
+                span.set_tag(
+                    COVERAGE_TAG_NAME,
+                    '{"files": [{"filename": "test_cov.py", "segments": [[5, 0, 5, 0, -1]]}, '
+                    + '{"filename": "test_module.py", "segments": [[2, 0, 2, 0, -1]]}]}',
+                )
+                span.finish()
+                conn = t._writer._conn
+                t.shutdown()
+            assert conn.request.call_count == (2 if compat.PY3 else 1)
+            assert conn.request.call_args_list[0].args[1] == "api/v2/citestcycle"
+            assert (
+                b"svc-no-cov" in conn.request.call_args_list[0].args[2]
+            ), "requests to the cycle endpoint should include non-coverage spans"
+            if compat.PY3:
+                assert conn.request.call_args_list[1].args[1] == "api/v2/citestcov"
+                assert (
+                    b"svc-no-cov" not in conn.request.call_args_list[1].args[2]
+                ), "requests to the coverage endpoint should not include non-coverage spans"
```

Comment:
Is this true? I thought that in the coverage encoder we only encode the spans with coverage but no strict requirement on non-coverage spans: 
```
spans_with_coverage = [span for span in spans if COVERAGE_TAG_NAME in span.get_tags()]
```

Reply:
As far as I know, the `citestcov` intake only understands payloads encoded according to the coverage payload spec. That spec requires that spans have coverage information to be encoded, so as far as I know it makes sense to exclude spans that don't have that information.